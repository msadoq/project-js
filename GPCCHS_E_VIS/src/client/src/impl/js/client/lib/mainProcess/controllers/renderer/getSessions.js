import getLogger from 'common/log';
import { server } from '../../ipc';
import { getStore } from '../../../store/mainStore';
import { updateSessions } from '../../../store/actions/sessions';

const logger = getLogger('main:controllers:renderer:getSessions');

export default function (reply, event, payload, queryId) {
  server.requestSessions((err, sessions = []) => {
    if (err) {
      return logger.error(err);
    }

    // update sessions list in store
    getStore().dispatch(updateSessions(sessions));

    // inform renderer process that sessions are up to date
    reply({ event: 'runCallback', queryId });
  });
}
