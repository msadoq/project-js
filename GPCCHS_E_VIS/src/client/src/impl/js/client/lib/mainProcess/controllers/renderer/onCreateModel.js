import _cloneDeep from 'lodash/cloneDeep';
import { dirname } from 'path';
import { getStore } from '../../../store/mainStore';
import { getView } from '../../../store/selectors/views';
import { add } from '../../../store/actions/messages';
import { getPathByFilePicker } from '../../dialog';
import { saveViewAs } from '../../../common/documentManager';
import vivl from '../../../../VIVL/main';
import { getRootDir } from '../../../common/fmd';


const root = getRootDir();

export default function ({ viewId }) {
  const { getState, dispatch } = getStore();
  const { type, configuration, absolutePath } = getView(getState(), viewId);

  const folder = absolutePath ? dirname(absolutePath) : root;
  return getPathByFilePicker(folder, 'model', 'save', (err, path) => {
    if (path) {
      // Create a twin view without formula in entry Points
      const modelConfig = _cloneDeep(configuration);
      const structureType = vivl(type, 'structureType')();
      if (structureType === 'last') {
        modelConfig.entryPoints.forEach((ep) => {
          ep.connectedData.formula = ''; // eslint-disable-line no-param-reassign
        });
      } else if (structureType === 'range') {
        modelConfig.entryPoints.forEach((ep) => {
          ep.connectedDataX.formula = ''; // eslint-disable-line no-param-reassign
          ep.connectedDataY.formula = ''; // eslint-disable-line no-param-reassign
        });
      }

      saveViewAs(modelConfig, type, path, (errSaving) => {
        if (errSaving) {
          dispatch(add(viewId, 'danger', `Model unsaved ${errSaving}`));
        } else {
          dispatch(add(viewId, 'success', 'Model saved'));
        }
      });
    }
  });
}
