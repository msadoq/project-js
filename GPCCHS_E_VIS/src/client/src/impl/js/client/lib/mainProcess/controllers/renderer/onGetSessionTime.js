import getLogger from 'common/log';
import reply from 'common/ipc/reply';
import { server } from '../../ipc';

const logger = getLogger('main:controllers:renderer:onGetSessionTime');

export default function (queryId) {
  server.requestSessionTime(({ err, time }) => {
    if (err) {
      return logger.error(err);
    }

    reply(queryId, time);
  });
}
