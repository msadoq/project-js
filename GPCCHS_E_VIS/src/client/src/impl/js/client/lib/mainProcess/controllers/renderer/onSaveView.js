import { dirname } from 'path';
import { getStore } from '../../../store/mainStore';
import { getView } from '../../../store/selectors/views';
import { setModified, updatePath, updateAbsolutePath, setViewOid } from '../../../store/actions/views';
import { addOnce as addMessage } from '../../../store/actions/messages';
import { getPathByFilePicker } from '../../dialog';
import { saveViewAs } from '../../../common/documentManager';

import { getRootDir, isFmd, getRelativeFmdPath } from '../../../common/fmd';

const getPath = path => (isFmd(path) ? getRelativeFmdPath(path) : path);

const root = getRootDir();
const addViewError = (viewId, msg) => addMessage(viewId, 'danger', msg);

export default function ({ viewId, saveAs }) {
  const { getState, dispatch } = getStore();
  const { type, configuration, absolutePath } = getView(getState(), viewId);

  function oncePath(savingAbsolutePath) {
    saveViewAs(configuration, type, savingAbsolutePath, (err, oid) => {
      if (err) {
        return getStore().dispatch(addViewError(viewId, err));
      }
      if (saveAs !== savingAbsolutePath) {
        // only for 'Save as...' action
        dispatch(updatePath(viewId, getPath(savingAbsolutePath)));
        dispatch(updateAbsolutePath(viewId, savingAbsolutePath));
        if (oid) {
          dispatch(setViewOid(viewId, oid));
        }
      }

      dispatch(setModified(viewId, false));
      dispatch(addMessage(viewId, 'success', 'View saved'));
    });
  }

  if (saveAs) {
    return oncePath(saveAs);
  }

  const folder = absolutePath ? dirname(absolutePath) : root;
  return getPathByFilePicker(folder, 'view', 'save', (err, path) => oncePath(path));
}
