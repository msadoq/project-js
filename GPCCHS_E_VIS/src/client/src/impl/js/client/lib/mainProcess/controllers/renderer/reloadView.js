/**
 * Created by isis on 1/5/17.
 */
