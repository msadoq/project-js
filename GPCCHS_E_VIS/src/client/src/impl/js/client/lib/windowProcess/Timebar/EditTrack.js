import classnames from 'classnames';
import moment from 'moment';
import React, { Component, PropTypes } from 'react';
import { FormGroup } from 'react-bootstrap';
import styles from './Lefttab.css';

export default class EditTrack extends Component {

  static propTypes = {
    hideEditTimeline: PropTypes.func.isRequired,
    editTimeline: PropTypes.func.isRequired,
    timeline: PropTypes.object.isRequired,
    masterId: PropTypes.string,
  }

  constructor(...args) {
    super(...args);

    const { timeline } = this.props;
    this.state = {
      duration: moment.duration(timeline.offset)
    };
  }

  componentDidMount() {
    this.updateFields();
    document.addEventListener('keyup', this.hideAddTimeline);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ duration: moment.duration(nextProps.timeline.offset) });
  }

  componentDidUpdate() {
    this.updateFields();
  }

  componentWillUnmount() {
    document.removeEventListener('keyup', this.hideAddTimeline);
  }

  updateFields = () => {
    this.editTimelineIdEl.value = this.props.timeline.id;
    this.editTimelineMasterEl.checked = this.props.masterId === this.props.timeline.id;
    ['years', 'months', 'days', 'years', 'months', 'days', 'hours', 'minutes', 'seconds', 'milliseconds'].forEach((x) => {
      this[`${x}El`].value = this.state.duration[x]();
    });
  }

  hideAddTimeline = (e) => {
    e.preventDefault();
    // Hit escape
    if (e && e.keyCode && e.keyCode === 27) {
      this.props.hideEditTimeline();
    }
  }

  willEditTimeline = (e) => {
    e.preventDefault();
    const { editTimeline, timeline } = this.props;

    const duration = moment.duration();
    // Dynamically fullfilling inputs
    ['years', 'months', 'days', 'hours', 'minutes', 'seconds', 'milliseconds'].forEach((x) => {
      const val = parseInt(this[`${x}El`].value, 10) || 0;
      duration.add(val, x);
    });

    editTimeline(
      timeline.timelineId,
      this.editTimelineIdEl.value,
      duration.asMilliseconds(),
      this.editTimelineMasterEl.checked
    );
  }

  render() {
    const { timeline, masterId } = this.props;

    return (
      <form
        className={classnames('form-horizontal', styles.form, styles.editForm)}
        onSubmit={this.willEditTimeline}
        ref={(el) => { this.editTimelineFormEl = el; }}
      >
        <b>Edit a track :</b>
        <FormGroup className={styles.formGroup}>
          <b className={styles.labelFormControl}>Id</b>
          <input
            ref={(el) => { this.editTimelineIdEl = el; }}
            className={classnames('form-control', styles.formControl)}
            disabled={masterId === timeline.id}
          />
        </FormGroup>
        <FormGroup className={styles.formGroup}>
          <b className={styles.labelFormControl}>Master track</b>
          <input
            ref={(el) => { this.editTimelineMasterEl = el; }}
            type="checkbox"
            disabled={masterId === timeline.id}
          />
        </FormGroup>
        <FormGroup className={styles.formGroup}>
          <b className={styles.labelFormControl}>Offset Y M D</b>
          {
            ['years', 'months', 'days'].map((x, i) =>
              <input
                key={i}
                type="number"
                ref={(el) => { this[`${x}El`] = el; }}
                placeholder={x}
                className={classnames(styles.input, styles[`input_${x}`], 'form-control')}
              />
            )
          }
        </FormGroup>
        <FormGroup className={styles.formGroup}>
          <b className={styles.labelFormControl}>Offset H M S MS</b>
          {
            ['hours', 'minutes', 'seconds', 'milliseconds'].map((x, i) =>
              <input
                key={i}
                type="number"
                ref={(el) => { this[`${x}El`] = el; }}
                placeholder={x}
                className={classnames(styles.input, styles[`input_${x}`], 'form-control')}
              />
            )
          }
        </FormGroup>
        <FormGroup className={styles.formGroup}>
          <b className={styles.labelFormControl}>&nbsp;</b>
          <input
            type="submit"
            value="Edit track"
            className={classnames(styles.addTrackButton, 'btn-md', 'btn-primary')}
          />
        </FormGroup>
      </form>
    );
  }
}
