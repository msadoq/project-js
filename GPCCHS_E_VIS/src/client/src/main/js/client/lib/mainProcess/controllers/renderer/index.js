// ====================================================================
// HISTORY
// VERSION : 1.1.0 : : : 28/02/2017 : Initial version
// VERSION : 1.1.2 : DM : #3622 : 07/03/2017 : first draft on inspector: retrieve data from rtd on right-click
// VERSION : 1.1.2 : DM : #5822 : 16/03/2017 : resolve a rtd link in the inspector
// VERSION : 1.1.2 : DM : #5828 : 30/03/2017 : Add a F1 button in VIMA to open the docu wiki helper
// VERSION : 1.1.2 : DM : #5822 : 03/04/2017 : merge dev in working branch
// VERSION : 1.1.2 : DM : #5828 : 23/05/2017 : Move common/callback and common/ipc modules in client sub-component
// VERSION : 1.1.2 : DM : #6785 : 12/06/2017 : activate links in views .
// VERSION : 1.1.2 : DM : #5828 : 13/06/2017 : Move common/constants/ in client/ folder
// VERSION : 1.1.2 : DM : #5828 : 14/06/2017 : Cleanup client/ file organization and test helper modules
// VERSION : 1.1.2 : FA : ISIS-FT-2132 : 15/06/2017 : Ask to save before closing view or page
// VERSION : 1.1.2 : DM : #6700 : 16/06/2017 : Add store enhancers helpers code coverage and merge with dev
// VERSION : 1.1.2 : DM : #6700 : 26/06/2017 : Change windowProcess health management to dispatch a Redux action directly.
// VERSION : 1.1.2 : DM : #6785 : 29/06/2017 : Fix opening view link in a new page and read only path for link definition
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : First draft on catalog explorer
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : catalog explorer : open, close and browse items
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 06/07/2017 : Remove unused onSavePage ipc renderer to mainProcess controllers
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 06/07/2017 : Replace ipc openPage by askOpenPage redux action
// VERSION : 1.1.2 : FA : #7235 : 18/07/2017 : Clean old code in menuManager/IPC controller
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 18/07/2017 : Fix common/fmd, disable Links feature for now
// VERSION : 1.1.2 : DM : #6700 : 20/07/2017 : Remove obsolete onServerDebug interface, ipcs and model
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Clean IPC about openInspector .
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Clean IPC about FMD_GET/FMD_CREATE and RELOAD_SESSIONS/GET_SESSION_TIME
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Clean IPC about documents .
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Remove serverDebug ipc . .
// VERSION : 1.1.2 : DM : #6700 : 03/08/2017 : Merge branch 'dev' into dbrugne-data
// VERSION : 1.1.2 : FA : #7145 : 04/08/2017 : Clean IPC about opening wiki helper + create a store folder in mainProcess
// END-HISTORY
// ====================================================================

import constants from 'constants';

import handle from 'common/ipc/handle';
import reply from 'common/ipc/reply';

import onReduxCurrentState from './onReduxCurrentState';
import onReduxDispatch from './onReduxDispatch';
import onResolveLink from './onResolveLink';
import onResolveRteLink from './onResolveRteLink';
import onGetRteCatalogs from './onGetRteCatalogs';
import onGetRteItemNames from './onGetRteItemNames';
import onOpenRteItem from './onOpenRteItem';
import onFocusRteItem from './onFocusRteItem';
import onGetRteDomains from './onGetRteDomains';

const controller = {
  [constants.IPC_METHOD_REDUX_CURRENT_STATE]: (...args) => onReduxCurrentState(reply, ...args),
  [constants.IPC_METHOD_REDUX_DISPATCH]: onReduxDispatch,
  [constants.IPC_METHOD_RESOLVE_LINK]: onResolveLink,
  [constants.IPC_METHOD_GET_RTE_DOMAINS]: onGetRteDomains,
  [constants.IPC_METHOD_GET_RTE_CATALOGS]: onGetRteCatalogs,
  [constants.IPC_METHOD_GET_RTE_ITEM_NAMES]: onGetRteItemNames,
  [constants.IPC_METHOD_OPEN_RTE_ITEM]: onOpenRteItem,
  [constants.IPC_METHOD_FOCUS_RTE_ITEM]: onFocusRteItem,
  [constants.IPC_METHOD_RESOLVE_RTE_LINK]: onResolveRteLink,
};

export default (electronEvent, data) => handle(
  controller,
  data,
  response => electronEvent.sender.send('global', response)
);
