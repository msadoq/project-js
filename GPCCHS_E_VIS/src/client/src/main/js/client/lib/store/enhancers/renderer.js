/**
 * Renderer process store enhancer
 *
 * - intercepts dispatched actions => forwards to main process
 * - receives state diff from main process => apply
 */

export default function rendererStoreEnhancer(previousStoreCreator) {
  if (process.type !== 'renderer') {
    throw new Error('rendererStoreEnhancer used in a non-renderer process');
  }

  return function createStore(reducer, initialState) {
    const store = previousStoreCreator(reducer, initialState);

    // TODO : enhance
    const storeDotDispatch = store.dispatch;
    store.dispatch = (action) => {
      console.log('VIMA ACTION on rendererStoreEnhancer');
      storeDotDispatch(action);
    };

    return store;
  };
}
