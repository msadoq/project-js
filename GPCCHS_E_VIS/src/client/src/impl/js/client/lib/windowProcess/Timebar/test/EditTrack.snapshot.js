import React from 'react';
import renderer from 'react-test-renderer';
import TimebarFixture from '../../../../test/__mocks__/Timebar';
import TimelinesFixture from '../../../../test/__mocks__/Timelines';
import EditTrack from '../EditTrack';

const propsStub = {
  hideEditTimeline: () => null,
  onClose: () => null,
  editTimeline: () => null,
  timeline: TimelinesFixture[0],
  masterId: TimebarFixture.masterId,
  sessions: [],
};

test('EditTrack renders correctly', () => {
  const tree = renderer.create(
    <EditTrack
      {...propsStub}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
