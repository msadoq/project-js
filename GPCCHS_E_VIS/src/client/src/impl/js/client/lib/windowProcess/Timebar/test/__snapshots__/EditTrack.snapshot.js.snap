exports[`test EditTrack renders correctly 1`] = `
<form
  className="form-horizontal"
  onSubmit={[Function]}>
  <div
    className="formGroup form-group">
    <b
      className="labelFormControl">
      Id
    </b>
    <input
      className="form-control formControl"
      disabled={true} />
  </div>
  <div
    className="formGroup form-group">
    <b
      className="labelFormControl">
      Master track
    </b>
    <input
      disabled={true}
      type="checkbox" />
  </div>
  <div
    className="formGroup form-group">
    <b
      className="labelFormControl">
      Session
    </b>
    <select
      className="form-control formControl"
      defaultValue={0} />
  </div>
  <div
    className="formGroup form-group">
    <b
      className="labelFormControl">
      Offset Y M D
    </b>
    <input
      className="input input_years form-control"
      placeholder="years"
      type="number" />
    <input
      className="input input_months form-control"
      placeholder="months"
      type="number" />
    <input
      className="input input_days form-control"
      placeholder="days"
      type="number" />
  </div>
  <div
    className="formGroup form-group">
    <b
      className="labelFormControl">
      Offset H M S MS
    </b>
    <input
      className="input input_hours form-control"
      placeholder="hours"
      type="number" />
    <input
      className="input input_minutes form-control"
      placeholder="minutes"
      type="number" />
    <input
      className="input input_seconds form-control"
      placeholder="seconds"
      type="number" />
    <input
      className="input input_milliseconds form-control"
      placeholder="milliseconds"
      type="number" />
  </div>
  <div
    className="formGroup form-group">
    <b
      className="labelFormControl">
      Â 
    </b>
    <input
      className="addTrackButton btn-md btn-primary"
      type="submit"
      value="Edit track" />
  </div>
</form>
`;
